wpeverest/user-registration
# UR-1618 Refactor - Custom notice conditionally rendering system.

### All Submissions:






### Changes proposed in this Pull Request:
Previously, we handled notices and their conditions in separate functions for each notice. In this PR, I have refactored the notice-adding mechanism so that it is now easier to add a notice and its conditions in a single array. Additionally, I have added functionality to fetch and render active, high-priority, and condition-matched notices from the server's JSON file.

### How to test the changes in this Pull Request:

1. Add High Priority, active and matched conditions in json file.
2. Check that notices is displayed in your dashboard or not.
3. Also check that our exists notices like review, survey, allow-usage are displayed properly.
4. Also check all buttons are properly working or not.
5. Also we have now reopen_days and reopen_times so check this too.

Here is the required information regarding this.

1. Json File only value's can be modify if you modify key then it may not work as expected. So be careful while creating json file.
2. Highest Priority is 1 - Low priority can be 2,3,4,....
3. reopen_days should in string numeric value.
4. Version value should be like >2.0 , <2.0  , !=2.0 , =3.0 , <=3.0  , >=3.0
5. While version compare of products key will be slug of plugin or theme.

### Types of changes:




### Other information:




### Changelog entry

> UR-1618 Refactor - Custom notice conditionally rendering system.
